packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1381,6 +1382,42 @@ class Transform extends SingleChildRenderObjectWidget {
         assert(scale == null || (scaleX == null && scaleY == null), "If 'scale' is non-null then 'scaleX' and 'scaleY' must be left null"),
         transform = Matrix4.diagonal3Values(scale ?? scaleX ?? 1.0, scale ?? scaleY ?? 1.0, 1.0);
 
+  // Computes a rotation matrix for an angle in radians, attempting to keep rotations
+  // at integral values for angles of 0, π/2, π, 3π/2.
+  static Matrix4 _computeRotation(double radians) {
+    if (radians == 0.0) {
+      return Matrix4.identity();
+    }
+    final double sin = math.sin(radians);
+    if (sin == 1.0) {
+      return _createZRotation(1.0, 0.0);
+    }
+    if (sin == - 1.0) {
+      return _createZRotation(-1.0, 0.0);
+    }
+    if (math.cos(radians) == -1.0) {
+      return _createZRotation(0.0, -1.0);
+    }
+    return Matrix4.rotationZ(radians);
+  }
+
+  static Matrix4 _createZRotation(double sin, double cos) {
+    final Matrix4 result = Matrix4.identity();
+    result.storage[0] = cos;
```

Comment:
Since the matrix is identity, you only need 4 assignments of cos/sin here.

Reply:
Or, if you use `.zero()` like the `rotationZ()` factory then you are missing an assignment of storage[15] to 1.0.

Also, in either case, you can skip the 0.0 because both `.zero()` and `.identity()` set all of the other entries to 0 for you.

So,
- do get rid of all of the `= 0.0` lines
- Either switch to the `.zero()` factory and add `storage[15] = 1.0`, or get rid of the `= 1.0` lines as well.

Reply:
Done